deploy/kubernetes/elastic-agent-kustomize/default/README.md

```diff
@@ -0,0 +1,72 @@
+# Kustomize Templates
+
+The list below includes the official [kustomize](https://github.com/kubernetes-sigs/kustomize) templates to run them in Kubernetes:
+
+Agent Scenario | Description
+---- | ----
+[Elastic Agent managed - Default ](./elastic-agent-managed/) | Default Elastic Agent managed by Fleet setup. Kube-state-metrics (KSM) is installed automatically.
+[Elastic Agent standalone Default ](./elastic-agent-standalone/) | Default Standalone Elastic Agent setup. Kube-state-metrics (KSM) is installed automatically.
+
+## Using above templates
+
+Users can clone this repository to use the provided kustomize templates.
+
+For *Managed Elastic Agent*, please update the following secrets inside main kustomization.yaml:
+
+- ${fleet_url}: Fleet Server URL to enroll the Elastic Agent into. FLEET_URL can be found in Kibana, go to Management > Fleet > Settings
+- ${enrollment_token}: Elasticsearch API key used to [enroll Elastic Agents](https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html#fleet-enrollment-tokens) in Fleet.
+
+```yaml
+secretGenerator:
+    - name: elastic-agent-creds
+      literals:
+        - host=${fleet_url}
+        - api_key=${enrollment_token}
+```
+
+For *Standalone Elastic Agent*, please update the following secrets inside main [kustomization.yaml](./elastic-agent-managed/kustomization.yaml):
+
+- ${es_host}: The Elasticsearch host to communicate with
+- ${api_key}: The API Key with access privilleges to connect to Elasticsearch. See [create-api-key-standalone-agent](https://www.elastic.co/guide/en/fleet/current/grant-access-to-elasticsearch.html#create-api-key-standalone-agent)
+
+
+## Remote usage of kustomize templates
+
+Users can use following commands:
+
+Managed Elastic Agent:
+
+```bash
```

Comment:
Would this be the command our users would get from this Kibana page?

![image](https://github.com/elastic/elastic-agent/assets/113898685/b9e8ad57-7a10-4660-b6fc-8c67b6edd833)

Reply:
I think a more intuitive approach would be to change the `kustomization.yaml` file directly. So the user would see this file in Kibana:

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

namespace: kube-system

images:
  - name: docker.elastic.co/beats/elastic-agent
    newTag: ${IMAGE_TAG}

secretGenerator:
    - name: elastic-agent-creds
      literals:
        - host=${ES_HOST} 
        - enrollment_token=${ENROLLMENT_TOKEN} 

resources:
  - ./base
  - https://github.com/kubernetes/kube-state-metrics

patches:
- path: environmental-variables-remove.yaml
- target:
    group: apps
    version: v1
    kind: DaemonSet
    name: elastic-agent
  path: fleet-enrollment-token-patch.yaml
```

And all `${...}` variables should have literal values by that point.

I think the problem with that command is that it still requires us to evaluate `base64` of the variables and we end up not making use of the literal of the secret map. And then the `sed` is not easy to read.

Reply:
All of this will be replaced by onliner command and nothing will be visible to the user